xmake/rules/vala/xmake.lua

```diff
@@ -103,22 +99,48 @@ rule("vala.build")
                 table.insert(argv, path(headerfile))
             end
         end
-        local vapidir = target:data("vala.vapidir")
+
+        local vapidir = target:values("vala.vapidir")
         if vapidir then
             table.insert(argv, path(vapidir, function (p) return "--vapidir=" .. p end))
         end
-        local valaflags = target:data("vala.flags")
+
+        local valaflags = target:values("vala.flags")
         if valaflags then
             table.join2(argv, valaflags)
         end
-        table.insert(argv, path(sourcefile_vala))
+
+        -- iterating through source files,
+        -- otherwise valac would fail when compiling multiple files
+        for _, sourcefile in ipairs(sourcebatch.sourcefiles) do
+            -- if it's only a vala file
+            if path.extension(sourcefile) == ".vala" then
+                table.insert(argv, path(sourcefile))
+            end
+        end
+
+        batchcmds:show_progress(opt.progress, "${color.build.object}compiling.vala")
```

Comment:
no sourcefile name in progress text. and Incremental compilation does not work.

```console
bash-5.2$ xmake
[ 25%]: compiling.vala
[100%]: build ok, spent 0.28s

bash-5.2$ xmake
[ 25%]: compiling.vala
[100%]: build ok, spent 0.28s
```

Reply:
```console
$ cd staticlib

$ xmake
[ 20%]: compiling.vala
[ 40%]: archiving.release libmymath.a
[ 70%]: compiling.vala
[ 90%]: linking.release test
[100%]: build ok, spent 0.695s

$ xmake
[ 20%]: compiling.vala
[ 70%]: compiling.vala
[100%]: build ok, spent 0.517s
```

Reply:
Because, as I explained, valac has to be called each time, since it acts on all of `.vala` files

Same for lack of fileblname, it compiles all vala files together

Reply:
you can show file progress one-by-one in for-loop.

Reply:
That would be something like this though:
```
[50%] compiling vala: src/main.vala
[50%] compiling vala: src/other.vala
[50%] compiling vala: src/sub/mod.vala
...
```
And so on for all sourcefiles displayed at once

If that's fine I'll correct it to do that

Reply:
right, you can do it.

Reply:
Ok, I'll also try getting it to at least not recompile if no files been modified